app/services/notify_service.rb

```diff
@@ -76,7 +76,7 @@ def response_to_recipient?
   end
 
   def from_staff?
-    @notification.from_account.local? && @notification.from_account.user.present? && @notification.from_account.user.staff?
+    @notification.from_account.local? && @notification.from_account.user.present? && @notification.from_account.user_role&.overrides?(@recipient.user_role)
```

Comment:
Since there is no concept of "staff" anymore (and I can imagine roles being used for stuff like "patreon supporter" etc without any staff-like implications), using role priority is the only way I can see to re-implement the intended effect here.